scipy/stats/_stats.pyx

```diff
@@ -0,0 +1,174 @@
+import warnings
+import numpy as np
+from numpy cimport int64_t, uint64_t
+import scipy.special as special
+from collections import namedtuple
+
+cimport numpy as np
+cimport cython
+
+# Kendall's tau will be computed directly between these types.
+# Arrays of other types will be turned into a rank array using _toranks().
+
+ctypedef fused ordered0:
+    np.int32_t
+    np.int64_t
+    np.float32_t
+    np.float64_t
+
+ctypedef fused ordered1:
+    np.int32_t
+    np.int64_t
+    np.float32_t
+    np.float64_t
+
+@cython.wraparound(False)
+@cython.boundscheck(False)
+cdef inline uint64_t pairs(uint64_t l):
+    return (l * (l - 1)) // 2
+
+@cython.wraparound(False)
+@cython.boundscheck(False)
+cdef inline double stat0(uint64_t l):
+    return l * (l - 1.) * (2*l + 5)
+
+@cython.wraparound(False)
+@cython.boundscheck(False)
+cdef inline double stat1(uint64_t l):
+    return l * (l - 1.) * (l - 2)
+
+# Turn any array into an array of ranks.
+
+@cython.wraparound(False)
+@cython.boundscheck(False)
+def _toranks(x):
+    cdef long[:] perm = np.argsort(x, kind='quicksort')
+    cdef long[:] rank = np.ndarray(len(perm), dtype=long)
```

Comment:
Declare perm and rank contiguous and do not use long as dtype (it fails on Windows 64 where long is 32 bit)